etc/config/metadata/component-registries/elyra-airflow-directory-preconfig.json

```diff
@@ -0,0 +1,11 @@
+{
+  "display_name": "Airflow Preloaded Components - Directory",
+  "metadata": {
+    "description": "Preloaded directory-based components that are supported by Apache Airflow",
+    "runtime": "airflow",
+    "location_type": "Directory",
+    "categories": ["Preloaded Airflow"],
+    "paths": ["airflow"]
+  },
+  "schema_name": "component-registry"
+}
```

Comment:
There are a few changes needed here based on the new schemas introduced by BYO catalog types
```JSON
{
  "display_name": "Airflow Preloaded Components - Directory",
  "metadata": {
    "description": "Preloaded directory-based components that are supported by Apache Airflow",
    "runtime": "airflow",
    "categories": ["Preloaded Airflow"],
    "paths": ["airflow"]
  },
  "schema_name": "local-directory-catalog",
  "version": 1
}
```

Reply:
Don't we still need the [location_type](https://github.com/elyra-ai/elyra/blob/b54a24da56b47efb369d459ca6d69035bbe4f912/elyra/metadata/schemas/component-registry.json#L62)?

Also, should [component-registry](https://github.com/elyra-ai/elyra/blob/master/elyra/metadata/schemas/component-registry.json) be called component-catalog?

Reply:
> Don't we still need the location_type?

That is a function of the catalog connector now.  Each connector is aware of its location type.

> Also, should component-registry be called component-catalog?

This schema is deprecated as of this PR:
https://github.com/elyra-ai/elyra/blob/b54a24da56b47efb369d459ca6d69035bbe4f912/elyra/metadata/schemas/component-registry.json#L10

That said, we'd like to rename the _schemaspace_ from `component-registries` to `component-catalogs` but you can't rename schemaspaces (today).  Instead, we're likely going to introduce a new schemaspace and deprecate `component-registries` - which it (and the component-registry schema) should stay around for a release to properly handle migration.